src/filesystem.cpp

```diff
@@ -734,120 +688,133 @@ bool FileInputStream::read(byte *buffer, unsigned size)
     return false;
 }
 
-const std::string* LocalPath::editStringDirect() const
+bool LocalPath::empty() const
 {
-    // this function for compatibiltiy while converting to use LocalPath class.  TODO: phase out this function
-    return const_cast<std::string*>(&localpath);
+    return localpath.empty();
 }
 
-std::string* LocalPath::editStringDirect()
+void LocalPath::clear()
 {
-    // this function for compatibiltiy while converting to use LocalPath class.  TODO: phase out this function
-    return const_cast<std::string*>(&localpath);
+    localpath.clear();
 }
 
-bool LocalPath::empty() const
+void LocalPath::erase(size_t pos, size_t count)
 {
-    return localpath.empty();
+    localpath.erase(pos, count);
+}
+
+void LocalPath::truncate(size_t bytePos)
+{
+    localpath.resize(bytePos);
 }
 
-size_t LocalPath::lastpartlocal(const FileSystemAccess& fsaccess) const
+LocalPath LocalPath::leafName(separator_t localseparator) const
 {
-    return fsaccess.lastpartlocal(&localpath);
+    auto p = localpath.find_last_of(localseparator);
+    p = p == string::npos ? 0 : p + 1;
+    LocalPath result;
+    result.localpath = localpath.substr(p, localpath.size() - p);
+    return result;
```

Comment:
could something like be more efficient?
```suggestion
return LocalPath(localpath.substr(p, localpath.size() - p));
```

Reply:
Possibly marginally but nowadays with the move functionality, and/or return value optimization, there will be little or no difference.  But the main consideration is that if we have a constructor that takes a std::string, then there could be some confusion about whether that string is platform encoded already or not, and accidental usage of it.